sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -207,6 +224,15 @@ def get_final_get_url(self, pipeline_response):
 
         :rtype: str
         """
+        if (
```

Comment:
Ok thought about this a little more, and it's not exactly what we want. For instance, if the value is Location, we want to force to Location, while today we won't.

So should look like something:

```python
if self._lro_options.get(_LroOption.FINAL_STATE_VIA) == _FinalStateViaOption.LOCATION_FINAL_STATE and self._location_url:
   return self._location_url
elif (
            self._lro_options.get(_LroOption.FINAL_STATE_VIA)
            in [
                _FinalStateViaOption.AZURE_ASYNC_OPERATION_FINAL_STATE,
                _FinalStateViaOption.OPERATION_LOCATION_FINAL_STATE
            ]
  ):
   return None

# code as usual
```

This is more straigtforward, and address the initial need: if I want to force something, no matter what was on the wire I force it

Reply:
One question: should we always return the location url if final state via is listed as location. Meaning, if they specify that the final state will be from location url, should we always stick strongly to that, regardless of whether the location url is `None`. In that case we'd be fully trusting the "final state via" option, which can be fair since a user explicitly passed in that lro option

Reply:
Good questio, no we should not:
```
if self._lro_options.get(_LroOption.FINAL_STATE_VIA) == _FinalStateViaOption.LOCATION_FINAL_STATE \ # This is the option
    and self._location_url: # This exists
```